light-client/src/header/eth_headers.rs

```diff
@@ -116,14 +125,36 @@ impl ETHHeaders {
             ));
         }
         let mut last_error: Option<Error> = None;
-        let headers = &self.all[..self.all.len() - 2];
-        for (i, header) in headers.iter().enumerate() {
-            let child = &self.all[i + 1];
-            let grand_child = &self.all[i + 2];
-            match verify_finalized(header, child, grand_child) {
-                Err(e) => last_error = Some(e),
-                Ok(()) => {
-                    if i + 2 != self.all.len() - 1 {
+        for i in 0..self.all.len() - 2 {
+            let finalized = &self.all[i];
+
+            // child: descendant whose vote.TargetNumber == finalized.Number
+            for j in (i + 1)..self.all.len() - 1 {
+                let child = &self.all[j];
+
+                // Ensure the relation between child and finalized is collect
```

Comment:
```suggestion
                // Ensure the relation between child and finalized is correct
```

There are several other similar typos.